examples/tutorials/matmul/int4_weight_decompression_cmnts.cpp

```diff
@@ -0,0 +1,265 @@
+
+    return ok ? 0 : 1;
+}
+
+} // namespace
+
+// Floating point MatMul
+// Inputs:
+// - Shape: M, N, K
+// - Matrices A and B
+// Outputs:
+// - Matrix C
+void ref_compute_matmul_f32(int64_t M, int64_t N, int64_t K, int64_t G,
+        std::vector<float> &A_f32, std::vector<float> &B_f32,
+        std::vector<float> &zp_B_f32, std::vector<float> &sc_B,
+        std::vector<float> &C_f32) {
+    // Perform the GEMM operation
+    for (int m = 0; m < M; ++m) {
+        for (int n = 0; n < N; ++n) {
+            for (int k = 0; k < K; ++k) {
+                // Decompress the weight
+                int64_t idx1 = k * N + n;
+                int64_t idx2 = (k / G) * N + n;
+                float decompressed_B
+                        = (B_f32[idx1] - zp_B_f32[idx1]) * sc_B[idx2];
+                // Perform the multiplication and accumulation
+                C_f32[m * N + n] += A_f32[m * K + k] * decompressed_B;
+            }
+        }
+    }
+}
+
+// Create a MatMul primitive descriptor for the following op:
+// C_f32 = A_f32 * (B_s4 - zp_B) * sc_B[:]
+matmul::primitive_desc matmul_pd_create(
+        int64_t M, int64_t N, int64_t K, int64_t G, const engine &eng) {
+
+    memory::desc a_md({M, K}, memory::data_type::f32, {K, 1}); // M x K layout
+    memory::desc b_md({K, N}, memory::data_type::s4,
+            memory::format_tag::any); // K x N layout
+    memory::desc c_md({M, N}, memory::data_type::f32, {N, 1}); // M x N layout
+
+    // Create attributes and indicate that the alpha and zero points are
+    // runtime parameters
+    primitive_attr attr;
+    // Set scales with multiple scales along K and N dimensions and with groups
+    // along K.
+    attr.set_scales(DNNL_ARG_WEIGHTS,
+            /* mask */ (1 << 0) + (1 << 1), {G, 1}, memory::data_type::f32);
+
+    // Set zero points with s4 data type.
+    // The mask determines which dimensions the zero points are applied to.
+    // Current mask value (1 << 0) + (1 << 1) means zero points are applied
+    // both along K and N dimensions.
+    // Changing the mask value would alter the dimensions along which the zero
+    // points are applied. For example:
+    // - mask = (1 << 0) would apply zero points only along the K dimension.
+    // - mask = (1 << 1) would apply zero points only along the N dimension.
+    int mask = (1 << 0) + (1 << 1); // zero points both along K and N dimensions
+    memory::dims groups = {};
+    attr.set_zero_points(DNNL_ARG_WEIGHTS, mask, groups, memory::data_type::s4);
+
+    // Set fpmath mode with `apply_to_int=true` to apply fpmath mode behavior to
+    // integral primitives (in this example, matmul).
+    attr.set_fpmath_mode(fpmath_mode::f16, true);
+
+    // Create a MatMul primitive descriptor
+    return matmul::primitive_desc(eng, a_md, b_md, c_md, attr);
+}
+
+// Function to perform matrix multiplication with int4 weights decompression
+// using oneDNN --Rupka
```

Comment:
Please remove the name "Rupak" from this comments

Reply:
removed